cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultScheduleConfigurer.java

```diff
@@ -121,6 +121,8 @@ public void triggerOnPartitions(String datasetName, int numPartitions) {
 
   private void setSchedule(ScheduleCreationSpec schedule) {
     // setSchedule can not be called twice on the same configurer (semantics are not defined)
-    Preconditions.checkArgument(null == programSchedules.put(schedule.getName(), schedule));
+    Preconditions.checkArgument(null == programSchedules.put(schedule.getName(), schedule),
+                                "Attempted to configure a schedule with name '%s' that was already " +
+                                  "configured.", schedule.getName());
```

Comment:
Created a JIRA to address this as a follow up:
https://issues.cask.co/browse/CDAP-11476